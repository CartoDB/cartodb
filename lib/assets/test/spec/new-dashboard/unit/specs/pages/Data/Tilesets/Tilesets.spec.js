import Vuex from 'vuex';
import { createLocalVue, shallowMount } from '@vue/test-utils';
import TilesetsComponent from 'new-dashboard/pages/Data/Tilesets/Tilesets';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('Connections.vue', () => {
  beforeEach(() => {
  });

  describe('Errors', () => {
    it('should render error if user has not projects', () => {
      const store = createStore();
      const connectionsList = createTilesets(store);

      expect(connectionsList).toMatchSnapshot();
    });

    it('should render error getting 401', () => {
      const store = createStore();
      const connectionsList = createTilesets(store);
      connectionsList.vm.$data.error = { status: 401 };

      expect(connectionsList).toMatchSnapshot();
    });

    it('should render error getting 403', () => {
      const store = createStore({
        connectors: {
          loadingConnections: false,
          connections: []
        },
        tilesets: {
          loadingTilesets: false
        },
        config: {
          isFirstTimeViewingDashboard: false
        }
      });
      const connectionsList = createTilesets(store);
      connectionsList.vm.$data.error = { status: 403 };

      expect(connectionsList).toMatchSnapshot();
    });

    it('should render error getting 404 fetching datasets', () => {
      const store = createStore({
        connectors: {
          loadingConnections: false,
          projects: [{ id: 'project1', friendly_name: 'Project 1' }],
          connections: []
        },
        tilesets: {
          loadingTilesets: false
        }
      });

      const connectionsList = createTilesets(store);
      connectionsList.setMethods({ fetchDatasets: () => {} });

      connectionsList.vm.$data.project = { id: 'project1', label: 'Project 1' };
      connectionsList.vm.$data.error = { status: 404 };

      expect(connectionsList).toMatchSnapshot();
    });

    it('should render no data when no tilesets', () => {
      const store = createStore({
        connectors: {
          loadingConnections: false,
          projects: [{ id: 'project1', friendly_name: 'Project 1' }],
          bqDatasets: [{ id: 'dataset1' }],
          connections: []
        },
        tilesets: {
          loadingTilesets: false
        }
      });

      const connectionsList = createTilesets(store);
      connectionsList.setMethods({ fetchDatasets: () => {} });

      connectionsList.vm.$data.project = { id: 'project1', label: 'Project 1' };
      connectionsList.vm.$data.dataset = { id: 'dataset1', label: 'dataset1' };

      expect(connectionsList).toMatchSnapshot();
    });
  });

  describe('Render', () => {
    it('should render list properly', () => {
      const store = createStore({
        connectors: {
          loadingConnections: false,
          projects: [{ id: 'project1', friendly_name: 'Project 1' }],
          bqDatasets: [{ id: 'dataset1' }],
          connections: []
        },
        tilesets: {
          loadingTilesets: false,
          tilesets: {
            result: [{
              id: 1,
              name: 'tile1'
            }, {
              id: 2,
              name: 'tile2'
            }, {
              id: 3,
              name: 'tile3'
            }]
          }
        }
      });

      const connectionsList = createTilesets(store);
      connectionsList.setMethods({ fetchDatasets: () => {}, fetchTilesets: () => {} });

      connectionsList.vm.$data.project = { id: 'project1', label: 'Project 1' };
      connectionsList.vm.$data.dataset = { id: 'dataset1', label: 'dataset1' };

      expect(connectionsList).toMatchSnapshot();
    });

    it('should render list with pagination properly', () => {
      const store = createStore({
        connectors: {
          loadingConnections: false,
          projects: [{ id: 'project1', friendly_name: 'Project 1' }],
          bqDatasets: [{ id: 'dataset1' }],
          connections: []
        },
        tilesets: {
          loadingTilesets: false,
          tilesets: {
            total: 25,
            result: [{
              id: 1,
              name: 'tile1'
            }, {
              id: 2,
              name: 'tile2'
            }, {
              id: 3,
              name: 'tile3'
            }]
          }
        }
      });

      const connectionsList = createTilesets(store);
      connectionsList.setMethods({ fetchDatasets: () => {}, fetchTilesets: () => {} });

      connectionsList.vm.$data.project = { id: 'project1', label: 'Project 1' };
      connectionsList.vm.$data.dataset = { id: 'dataset1', label: 'dataset1' };

      expect(connectionsList).toMatchSnapshot();
    });
  });
});

function createStore (customStoreData) {
  const storeData = {
    connectors: {
      loadingConnections: false,
      connections: []
    },
    tilesets: {
      loadingTilesets: false
    },
    config: {
      isFirstTimeViewingDashboard: false
    }
  };

  const storeInstance = new Vuex.Store({
    state: customStoreData || storeData,
    actions: {
      fetchBQDatasetsList: jest.fn()
    }
  });

  return storeInstance;
}

function createTilesets (store, propsData) {
  return shallowMount(TilesetsComponent, {
    mocks: {
      $t: key => key,
      $router: {
        push: jest.fn()
      }
    },
    store,
    localVue,
    propsData
  });
}
