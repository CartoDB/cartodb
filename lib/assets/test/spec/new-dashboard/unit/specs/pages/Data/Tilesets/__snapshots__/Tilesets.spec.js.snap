// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connections.vue Errors should render error getting 401 1`] = `
<section class="u-mt--64">
  <div class="container grid">
    <div class="full-width">
      <sectiontitle-stub class="header grid-cell"><template></template> <template></template></sectiontitle-stub>
      <div class="grid-cell grid-cell--col12">
        <div class="u-flex u-mt--20 u-mb--36">
          <div class="u-mr--28">
            <div class="text is-small is-semibold">
              Select Project
            </div>
            <dropdowncomponent-stub placeholder="TilesetsPage.projectPlaceholder" elements="" showcreate="true" type="text"></dropdowncomponent-stub>
          </div>
          <div class="dropdown-disabled">
            <div class="text is-small is-semibold">
              Project datasets
            </div>
            <dropdowncomponent-stub placeholder="Select dataset" elements="" type="text"></dropdowncomponent-stub>
          </div>
        </div>
        <!---->
        <div class="u-flex u-pt--48 u-pb--48 u-pl--32 u-pr--32 empty-list"><img src="../../../assets/icons/tilesets/tileset-error.svg">
          <div class="u-ml--32">
            <div class="text is-body is-semibold u-mb--12">
              TilesetsPage.errorTitle
            </div>
            <div class="text is-caption u-mb--16">TilesetsPage.errorSubtitle</div>
            <div class="is-small">
              <div class="u-mt--12 text is-txtMidGrey is-semibold">Error info:</div>
              <div class="u-mt--12 u-flex text is-semibold"><input type="text" readonly="readonly" class="code u-flex__grow--1 is-code">
                <div class="u-ml--4 copy"><img svg-inline="" src="../../../assets/icons/catalog/copy.svg"></div>
              </div>
            </div>
          </div>
        </div>
        <router-view></router-view>
      </div>
    </div>
  </div>
</section>
`;

exports[`Connections.vue Errors should render error getting 403 1`] = `
<section class="u-mt--64">
  <div class="container grid">
    <div class="full-width">
      <sectiontitle-stub class="header grid-cell"><template></template> <template></template></sectiontitle-stub>
      <div class="grid-cell grid-cell--col12">
        <div class="u-flex u-mt--20 u-mb--36">
          <div class="u-mr--28">
            <div class="text is-small is-semibold">
              Select Project
            </div>
            <dropdowncomponent-stub placeholder="TilesetsPage.projectPlaceholder" elements="" showcreate="true" type="text"></dropdowncomponent-stub>
          </div>
          <div class="dropdown-disabled">
            <div class="text is-small is-semibold">
              Project datasets
            </div>
            <dropdowncomponent-stub placeholder="Select dataset" elements="" type="text"></dropdowncomponent-stub>
          </div>
        </div>
        <!---->
        <div class="u-flex u-pt--48 u-pb--48 u-pl--32 u-pr--32 empty-list"><img src="../../../assets/icons/tilesets/tileset-error.svg">
          <div class="u-ml--32">
            <div class="text is-body is-semibold u-mb--12">
              TilesetsPage.errorTitle
            </div>
            <div class="text is-caption u-mb--16">TilesetsPage.errorSubtitle</div>
            <div class="is-small">
              <div class="u-mt--12 text is-txtMidGrey is-semibold">Error info:</div>
              <div class="u-mt--12 u-flex text is-semibold"><input type="text" readonly="readonly" class="code u-flex__grow--1 is-code">
                <div class="u-ml--4 copy"><img svg-inline="" src="../../../assets/icons/catalog/copy.svg"></div>
              </div>
            </div>
          </div>
        </div>
        <router-view></router-view>
      </div>
    </div>
  </div>
</section>
`;

exports[`Connections.vue Errors should render error getting 404 fetching datasets 1`] = `
<section class="u-mt--64">
  <div class="container grid">
    <div class="full-width">
      <sectiontitle-stub class="header grid-cell"><template></template> <template></template></sectiontitle-stub>
      <div class="grid-cell grid-cell--col12">
        <div class="u-flex u-mt--20 u-mb--36">
          <div class="u-mr--28">
            <div class="text is-small is-semibold">
              Select Project
            </div>
            <dropdowncomponent-stub placeholder="TilesetsPage.projectPlaceholder" elements="[object Object]" showcreate="true" type="text" value="[object Object]"></dropdowncomponent-stub>
          </div>
          <div class="">
            <div class="text is-small is-semibold">
              Project datasets
            </div>
            <dropdowncomponent-stub placeholder="Select dataset" elements="" type="text"></dropdowncomponent-stub>
          </div>
        </div>
        <!---->
        <div class="u-flex u-pt--48 u-pb--48 u-pl--32 u-pr--32 empty-list"><img src="../../../assets/icons/tilesets/tileset-error.svg">
          <div class="u-ml--32">
            <div class="text is-body is-semibold u-mb--12">
              TilesetsPage.notFoundTitle
            </div>
            <div class="text is-caption u-mb--16">TilesetsPage.notFoundSubtitle</div>
          </div>
        </div>
        <router-view></router-view>
      </div>
    </div>
  </div>
</section>
`;

exports[`Connections.vue Errors should render error if user has not projects 1`] = `
<section class="u-mt--64">
  <div class="container grid">
    <div class="full-width">
      <sectiontitle-stub class="header grid-cell"><template></template> <template></template></sectiontitle-stub>
      <div class="grid-cell grid-cell--col12">
        <div class="u-flex u-mt--20 u-mb--36">
          <div class="u-mr--28">
            <div class="text is-small is-semibold">
              Select Project
            </div>
            <dropdowncomponent-stub placeholder="TilesetsPage.projectPlaceholder" elements="" showcreate="true" type="text"></dropdowncomponent-stub>
          </div>
          <div class="dropdown-disabled">
            <div class="text is-small is-semibold">
              Project datasets
            </div>
            <dropdowncomponent-stub placeholder="Select dataset" elements="" type="text"></dropdowncomponent-stub>
          </div>
        </div>
        <!---->
        <div class="u-flex u-pt--48 u-pb--48 u-pl--32 u-pr--32 empty-list"><img src="../../../assets/icons/tilesets/tileset-error.svg">
          <div class="u-ml--32">
            <div class="text is-body is-semibold u-mb--12">
              TilesetsPage.errorTitle
            </div>
            <div class="text is-caption u-mb--16">TilesetsPage.errorSubtitle</div>
            <!---->
          </div>
        </div>
        <router-view></router-view>
      </div>
    </div>
  </div>
</section>
`;

exports[`Connections.vue Errors should render no data when no tilesets 1`] = `
<section class="u-mt--64">
  <div class="container grid">
    <div class="full-width">
      <sectiontitle-stub class="header grid-cell"><template></template> <template></template></sectiontitle-stub>
      <div class="grid-cell grid-cell--col12">
        <div class="u-flex u-mt--20 u-mb--36">
          <div class="u-mr--28">
            <div class="text is-small is-semibold">
              Select Project
            </div>
            <dropdowncomponent-stub placeholder="TilesetsPage.projectPlaceholder" elements="[object Object]" showcreate="true" type="text" value="[object Object]"></dropdowncomponent-stub>
          </div>
          <div class="">
            <div class="text is-small is-semibold">
              Project datasets
            </div>
            <dropdowncomponent-stub placeholder="Select dataset" elements="[object Object]" type="text" value="[object Object]"></dropdowncomponent-stub>
          </div>
        </div>
        <!---->
        <div class="u-flex u-pt--48 u-pb--48 u-pl--32 u-pr--32 u-flex__align--start empty-list"><img src="../../../assets/icons/tilesets/tileset-empty.svg">
          <div class="u-ml--32">
            <div class="text is-body is-semibold u-mb--12">
              TilesetsPage.noAvailableTitle
            </div>
            <div class="text is-caption u-mb--16">TilesetsPage.noAvailableSubtitle</div>
            <!---->
          </div>
        </div>
        <router-view></router-view>
      </div>
    </div>
  </div>
</section>
`;

exports[`Connections.vue Render should render list properly 1`] = `
<section class="u-mt--64">
  <div class="container grid">
    <div class="full-width">
      <sectiontitle-stub class="header grid-cell"><template></template> <template></template></sectiontitle-stub>
      <div class="grid-cell grid-cell--col12">
        <div class="u-flex u-mt--20 u-mb--36">
          <div class="u-mr--28">
            <div class="text is-small is-semibold">
              Select Project
            </div>
            <dropdowncomponent-stub placeholder="TilesetsPage.projectPlaceholder" elements="[object Object]" showcreate="true" type="text" value="[object Object]"></dropdowncomponent-stub>
          </div>
          <div class="">
            <div class="text is-small is-semibold">
              Project datasets
            </div>
            <dropdowncomponent-stub placeholder="Select dataset" elements="[object Object]" type="text" value="[object Object]"></dropdowncomponent-stub>
          </div>
        </div>
        <div class="grid__head--sticky">
          <tilesetlistheader-stub issortable="true"></tilesetlistheader-stub>
        </div>
        <ul>
          <li class="tileset-item">
            <tilesetlistcard-stub tileset="[object Object]" canhover="true" storeactiontype="tilesets" class="tileset-item"></tilesetlistcard-stub>
          </li>
          <li class="tileset-item">
            <tilesetlistcard-stub tileset="[object Object]" canhover="true" storeactiontype="tilesets" class="tileset-item"></tilesetlistcard-stub>
          </li>
          <li class="tileset-item">
            <tilesetlistcard-stub tileset="[object Object]" canhover="true" storeactiontype="tilesets" class="tileset-item"></tilesetlistcard-stub>
          </li>
        </ul>
        <!---->
        <!---->
        <router-view></router-view>
      </div>
    </div>
  </div>
</section>
`;

exports[`Connections.vue Render should render list with pagination properly 1`] = `
<section class="u-mt--64">
  <div class="container grid">
    <div class="full-width">
      <sectiontitle-stub class="header grid-cell"><template></template> <template></template></sectiontitle-stub>
      <div class="grid-cell grid-cell--col12">
        <div class="u-flex u-mt--20 u-mb--36">
          <div class="u-mr--28">
            <div class="text is-small is-semibold">
              Select Project
            </div>
            <dropdowncomponent-stub placeholder="TilesetsPage.projectPlaceholder" elements="[object Object]" showcreate="true" type="text" value="[object Object]"></dropdowncomponent-stub>
          </div>
          <div class="">
            <div class="text is-small is-semibold">
              Project datasets
            </div>
            <dropdowncomponent-stub placeholder="Select dataset" elements="[object Object]" type="text" value="[object Object]"></dropdowncomponent-stub>
          </div>
        </div>
        <div class="grid__head--sticky">
          <tilesetlistheader-stub issortable="true"></tilesetlistheader-stub>
        </div>
        <ul>
          <li class="tileset-item">
            <tilesetlistcard-stub tileset="[object Object]" canhover="true" storeactiontype="tilesets" class="tileset-item"></tilesetlistcard-stub>
          </li>
          <li class="tileset-item">
            <tilesetlistcard-stub tileset="[object Object]" canhover="true" storeactiontype="tilesets" class="tileset-item"></tilesetlistcard-stub>
          </li>
          <li class="tileset-item">
            <tilesetlistcard-stub tileset="[object Object]" canhover="true" storeactiontype="tilesets" class="tileset-item"></tilesetlistcard-stub>
          </li>
        </ul>
        <pagination-stub page="1" numpages="3"></pagination-stub>
        <!---->
        <router-view></router-view>
      </div>
    </div>
  </div>
</section>
`;
